### DISCUSSION TITLE
How to SKIP a failed task?

### DISCUSSION THREAD

**Discussion by cesar-moya** [Netflix/conductor#2214] [Q&A]
Currently you can only SKIP a future task by using the following API: https://netflix.github.io/conductor/apispec/#skip-task, but this doesn't let you SKIP a task that is already in FAILED state.
However, we have use cases where we need to SKIP a task that is in FAILED state (hence, continuing the workflow), looking at the API it seems the only operations supported on a workflow that failed because of a task failure are:
POST /workflow/{workflowId}/retry and POST /workflow/{workflowId}/rerun.

Are there any plans to support skipping a failed task by id, continuing the workflow, and keeping all of the execution history intact? If not, are there any suggestions as to how we could accomplish this?

Thanks,

**Comment by ravisund** [comment]
+1, we also need this feature.

**Comment by kishorebanala** [comment]
Hey @cesar-moya, Have you considered marking this task as Optional? Also, can you provide more details about how you plan on using this feature? 

It'd also help to have an example, particularly specifying what state the workflow and task to be skipped would be in, when trying to be skipped.

**Reply by AkashShuklaGithub** [reply]
@kishorebanala  Marking a task Optional automatically skips the failed task, without failing the workflow.    
Do we have any approach where task failure does fail the workflow, and after some manual remediation on the downstream services, the failed task can be skipped by a rest api call.

**Comment by ravisund** [comment]
We also need this functionality. Below is our use case,

When a workflow is failed, sometimes the failure will be manually remediated. Our tasks are not idempotent.  In such case, we need to skip the failed task if it was manually remediated. How do we achieve this in conductor ?
Below are some options
Option 1)  Skip Task API will be enhanced to handle this case.
Option 2) Retry Task API can accept additional input (similar to rerun API), this input will be populated into Task's Input. Now it is upto the Task implementer's responsibility to implement the skip functionality based on the input value. 

The reason, we can't use the current rerun API is, the rerun will reset (or clear all the previous execution details , e.g. we will lose the start & end timestamps of previous attempt).

**Comment by kishorebanala** [comment]
@AkashShuklaGithub @ravisund we've considered this, but we're concerned about the side effects, eg. when the downstream tasks rely on the output from this failed task. A thorough approach would be able to verify and validate these side effects. Given the returns, we've bucketed it as low priority. If anyone in the community has bandwidth to take this up, please open a RFC with a proposal.
